api/v1alpha1/proxmoxmachine_types.go

```diff
@@ -224,10 +224,20 @@ type NetworkDevice struct {
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=65520
 	MTU *uint16 `json:"mtu,omitempty"`
+
+	// DHCP4 indicates that if DHCP should be used to assign IPv4 addresses.
+	// DHCP4 enforce device to use DHCP despite the config set in cluster.spec.ipv4Config.
+	// +optional
+	DHCP4 bool `json:"dhcp4,omitempty"`
```

Comment:
Why is in the base network device? All permuations of DHCP and IPAM on the default device should be configured in the cluster network config.

This also has the same pointer issue as the cluster dhcp setting.